app/core/nep5.js

```diff
@@ -20,10 +22,16 @@ const getTokenEntry = ((): Function => {
   })
 })()
 
-export const getDefaultTokens = (): Array<TokenItemType> => {
+export const getDefaultTokens = async (): Promise<Array<TokenItemType>> => {
   const tokens = []
 
-  tokens.push(...map(TOKENS, (scriptHash, symbol) => getTokenEntry(symbol, scriptHash, MAIN_NETWORK_ID)))
+  // Prevent duplicate requests here
+  if (!fetchedTokens) {
```

Comment:
This will prevent tokens from being updated if list is updated after it was already fetched (meaning that list will be updated only after wallet restart after it's fetched for the first time).

Reply:
I did that on purpose thinking it would be ok to fetch once per login but we can change that

Reply:
Just wanted to mention. Not sure if we should change this at this point because fetching of blacklisted address and coinmarketcap prices for example are working in the same way currently. However we probably should address this somehow in future (v2).